### ISSUE TITLE
Adding the sphinx-gallery Binder button to Matplotlib

### ISSUE THREAD

**Issue by choldgraf** [jupyterhub/mybinder.org-deploy#298]
Now that Sphinx Gallery has a Binder button prototype in there, I'd like to start making PRs to add it to websites that already use sphinx-gallery. I was going to start with Matplotlib, since it has the largest reach. However, looking at the numbers I'm a bit worried that we'll end up saturating our traffic.

I took a look at the matplotlib daily traffic over the last few months only for pages within /gallery/, and it hovers somewhere around 10,000 users a day. If we assume that 1% of people actually click on a Binder link within those examples, that's still 100 extra users per day.

Currently we're hovering at around 800 hits/day to mybinder.org. That means if we start getting an extra 100 a day from another source, that's a 12% increase from a single source.

So, I wanted to open this issue to make sure we're cool with that. It's kind of a double-edged sword, since obviously a 12% bump in traffic would be awesome. But I want to make sure it doesn't bring the cluster down to a halt...

what do folks think?

**Comment by yuvipanda** [comment]
I think we'll be fine!

**Comment by betatim** [comment]
I also think we will be fine. Would also go live on the "master" version of the docs for matplotlib first or also on the docs for the "released" version? I assume there is less traffic on the master version.

Do you have good relations with the matplotlib guys? Just in case we decide that something needs patching "quickly".

**Comment by choldgraf** [comment]
> Do you have good relations with the matplotlib guys? Just in case we decide that something needs patching "quickly"

Yep, I refactored almost all of their documentation last year, so shouldn't be too big a deal to get a patch in there. And good point about the devdocs, it'll be a bit of time before the non dev version of the site gets updated

**Comment by yuvipanda** [comment]
Did this get added, @choldgraf? Esp. with the recent '100 launches per repo' setup I'm very happy to have this go forward.

**Comment by choldgraf** [comment]
It is now in `master` on sphinx-gallery, and whenever we cut a new release I'll make a PR to include it in the devdocs! Rest assured I will ping when this happens!

**Comment by choldgraf** [comment]
hey folks - that sphinx-gallery PR finally got merged! I'm planning to make a PR to the matplotlib docs to include Binder links in the examples pages...are folks OK with that? If not, please do speak now!

**Comment by betatim** [comment]
Sounds good to me. How quickly would it get deployed/can we ask for it to go live at a time that is good for the team? Can we enable it on a low traffic part of the matplotlib docs first (maybe the `/latest/` pages instead of those of the current release)?

**Comment by choldgraf** [comment]
> For the PR will you explain to the matplotlib people how experimental mybinder.org is, the restrictions, availability of the team, etc? Just to make sure they aren't disappointed or annoyed. I'd rather under-promise and over-deliver (or how ever that saying goes)

hmmm - this makes me feel a bit more like we should just hold off on this. Matplotlib gets like 30,000 visits to the docs a _day_, and if we think that'll overwhelm Binder I'd rather be confident it will work and _then_ flip the switch, rather than the other way around.

Is the concern HTTP hits on the JupyterHub? I'm curious what @minrk thinks in terms of how likely it'd overwhelm the service...

**Comment by betatim** [comment]
I don't know if it will overwhelm mybinder.org or not. However I confidently predict that there will be some kind of hiccup, no matter how ready we think we are. So I think we should try and prepare several levels of "turning it off again", including right at the source. That way we don't put ourselves under pressure to fix things live. No one even knows how many of the 30000 visitors would even click on a binder link, which makes it hard to know if we are ready.

One thing to do is to only show the mybinder.org button to a small fraction of the matplotlib.org traffic at the start and step it up gradually.

**Comment by yuvipanda** [comment]
We should do this now :D I don't see it in like, https://matplotlib.org/stable/gallery/lines_bars_and_markers/barchart.html#sphx-glr-gallery-lines-bars-and-markers-barchart-py. How can we push this along, @choldgraf?

**Comment by choldgraf** [comment]
Mmm i think we'd need to open an issue or a PR in matplotlib. I'm not sure where their thinking is on binder these days, so not sure of they'd be receptive to it or not. I don't have much bandwidth to really push something if they're not an immediate 👍 though :-\
